pandas/tests/frame/indexing/test_mask.py

```diff
@@ -150,3 +150,11 @@ def test_mask_inplace_no_other():
     df.mask(cond, inplace=True)
     expected = DataFrame({"a": [np.nan, 2], "b": ["x", np.nan]})
     tm.assert_frame_equal(df, expected)
+
+
+def test_mask_with_NA():
+    df = DataFrame({"A": [0, 1, 2]})
+    cond = Series([-1, 1, None]).convert_dtypes() < 0
```

Comment:
```suggestion
    cond = Series([True, False, pd.NA], dtype=pd.BooleanDtype())

```

Reply:
Thank you, @WillAyd , for the helpful suggestion. I've implemented the change.